packages/components/src/components/vertical-tab/vertical-tab-headers.jsx

```diff
@@ -5,110 +5,95 @@ import VerticalTabHeader from './vertical-tab-header.jsx';
 import VerticalTabHeaderGroup from './vertical-tab-header-group.jsx';
 import VerticalTabHeaderTitle from './vertical-tab-header-title.jsx';
 
-class VerticalTabHeaders extends React.PureComponent {
-    ref = React.createRef();
-    headers = [];
-    state = { top: 0, should_skip_animation: false };
+const VerticalTabHeaders = ({
+    className,
+    header_title,
+    is_floating,
+    is_routed,
+    item_groups,
+    items,
+    onChange,
+    selected,
+}) => {
+    const ref = React.useRef(null);
+    const [top, setTop] = React.useState(0);
+    const [should_skip_animation, setShouldSkipAnimation] = React.useState(false);
 
-    componentDidMount() {
-        this.repositionActiveHighlighter();
-    }
```

Comment:
Don't we need a corresponding `effect` for this? Same for `componentDidUpdate`.

Reply:
We actually don't need this function anymore as it was used to set the both the states `top` and `should_skip_animation`. Instead `top` is being set with [this `useEffect`](https://github.com/binary-com/deriv-app/pull/2233/files/bfdc90d7f3de4ac3bbfa81ca22e9a47987bd02ba#diff-3658191ed864a4c75732cb1c6472ce89199b66b9335dec6ff4869bb24a855fcbR22-R24).

I traced the code for `should_skip_animation` it was using `false` as default value and it is only being set by `onToggle` property in `VerticalTabHeaderGroup` Component so I pass `setShouldSkipAnimation` to that component instead. Please refer to [this](https://github.com/binary-com/deriv-app/pull/2233/files/bfdc90d7f3de4ac3bbfa81ca22e9a47987bd02ba#diff-3658191ed864a4c75732cb1c6472ce89199b66b9335dec6ff4869bb24a855fcbR57) and [this](https://github.com/binary-com/deriv-app/pull/2233/files/bfdc90d7f3de4ac3bbfa81ca22e9a47987bd02ba#diff-ffda9f6f88e2f01e7b2b23f70ab8cbfeba2c345fe0de2de39865b560c8103464R23-R25)